skt/state_file.py

```diff
@@ -85,3 +87,7 @@ def update(cfg, state_updates):
     except IOError as exception:
         logging.error("Failed to update state file: %s", exception)
         raise exception
+
+    # update in memory copy
+    state = new_state.copy()
```

Comment:
Agreed, noticed when I actually did circleci testing.  Changing api to return the new_state, instead of playing with clear() and update().  More noisy, but more obvious?